### PR TITLE
Adding nvme-cli selftests

### PR DESCRIPTION
**Pull request by narasimhan-v**
Adding nvme-cli selftests

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

File: io/disk/ssd/nvme_cli_selftests.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        msg.append('}')
+        with open('config.json', 'w') as config_file:
+            config_file.write("\n".join(msg))
+        process.system("cat config.json")
+
+    def test_selftests(self):
+        """
+        Runs the selftests on the device.
+        """
+        ret = process.run('make run', shell=True, ignore_status=True)
```
Comment (PraveenPenguin) [inline io/disk/ssd/nvme_cli_selftests.py]: can `build.run_make `used here 

from avocado.utils import build

```
 for line in build.run_make(os.path.join(self.nvme_dir, 'tests'), extra_args='run').stderr.splitlines():
        if 'FAIL:' in line:
 
```

Comment (narasimhan-v) [inline io/disk/ssd/nvme_cli_selftests.py]: Done, force pushed :)

**Comment by PraveenPenguin** [comment]
@narasimhan-v can you please provide job as my test system don't have nvme device

**Comment by narasimhan-v** [comment]
```
2017-07-04 07:52:22,677 process          L0479 DEBUG| [stderr] make: *** [run] Error 1
2017-07-04 07:52:22,677 process          L0499 INFO | Command 'make run' finished with 2 after 675.669334888s
2017-07-04 07:52:22,679 stacktrace       L0039 ERROR|
2017-07-04 07:52:22,679 stacktrace       L0042 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_framework-51.0-py2.7.egg/avocado/core/test.py:596
2017-07-04 07:52:22,680 stacktrace       L0045 ERROR| Traceback (most recent call last):
2017-07-04 07:52:22,681 stacktrace       L0045 ERROR|   File "/root/avocado-fvt-wrapper/tests/avocado-misc-tests/io/disk/ssd/nvme_cli_selftests.py", line 85, in test_selftests
2017-07-04 07:52:22,681 stacktrace       L0045 ERROR|     self.fail("Some tests failed. Details below:\n%s" % "\n".join(err))
2017-07-04 07:52:22,681 stacktrace       L0045 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_framework-51.0-py2.7.egg/avocado/core/test.py", line 762, in fail
2017-07-04 07:52:22,682 stacktrace       L0045 ERROR|     raise exceptions.TestFail(message)
2017-07-04 07:52:22,682 stacktrace       L0045 ERROR| TestFail: Some tests failed. Details below:
2017-07-04 07:52:22,682 stacktrace       L0045 ERROR| TestNVMeCompareCmd
2017-07-04 07:52:22,683 stacktrace       L0045 ERROR| TestNVMeCreateMaxNS
2017-07-04 07:52:22,683 stacktrace       L0045 ERROR| TestNVMeFlushCmd
2017-07-04 07:52:22,683 stacktrace       L0045 ERROR| TestNVMeGetMandetoryFeatures
2017-07-04 07:52:22,683 stacktrace       L0045 ERROR| TestNVMeReadWriteTest
2017-07-04 07:52:22,684 stacktrace       L0045 ERROR| TestNVMeSmartLogCmd
2017-07-04 07:52:22,684 stacktrace       L0045 ERROR| TestNVMeUncor
2017-07-04 07:52:22,684 stacktrace       L0045 ERROR| TestNVMeWriteZeros
2017-07-04 07:52:22,685 stacktrace       L0046 ERROR|
```

**Pull request merged**